src/main/java/com/passbook/sparkeighteen/service/UserService.java

```diff
@@ -0,0 +1,40 @@
+package com.passbook.sparkeighteen.service;
+
+import com.passbook.sparkeighteen.peristence.POJO.SignUpRequest;
+import com.passbook.sparkeighteen.peristence.POJO.SignUpResponse;
+import com.passbook.sparkeighteen.peristence.entity.UserEntity;
+import com.passbook.sparkeighteen.peristence.repository.UserRepository;
+import lombok.NonNull;
+import org.springframework.stereotype.Service;
+
+
+import java.util.Optional;
+
+@Service
+public class UserService {
+
+    private UserRepository userRepository;
+
+    public UserService(final UserRepository userRepository) {
```

Comment:
Why did we use constructor injection.
What are the alternative injection strategy.

Reply:
It is a type of Spring Dependency Injection, where object's constructor is used to inject dependencies. This type of injection is safer as the objects won't get created if the dependencies aren't available or dependencies cannot be resolved.

Alternatives injection 
1. Constructor-Based Injection
2. Setter Injection
3. Property-Based Injection